tests/test_estimation_basic_attributes.py

```diff
@@ -0,0 +1,183 @@
+            fstAttr, sndAttr = map(attrgetter(attrName), (eth1, eth2))
+            assert abs(fstAttr - sndAttr) <= delta, f"Attribute '{attrName}' differ: {fstAttr} {sndAttr}"
+        assert eth1.predominantEthnicity == eth2.predominantEthnicity
+
+    def test_correctness(self):
+        """
+        Test estimation correctness.
+        """
+        expectedEstimation = Estimation(20, 0, Eth(0, 0, 1, 0))
+        for estimationType in ("Age", "Gender", "Ethnicity"):
+            estimationFlag = f"estimate{estimationType}"
+            basicAttributeGetter: Callable[[BasicAttributes], Union[Ethnicities, float, None]] = attrgetter(
+                estimationType.lower()
+            )
+            expectedValue = getattr(expectedEstimation, estimationType)
+            with self.subTest(estimationType=estimationType):
+                singleValue = basicAttributeGetter(self.estimate(self._warp, **{estimationFlag: True}))
+                batchValue = basicAttributeGetter(self.estimateBatch([self._warp] * 2, **{estimationFlag: True})[0][0])
+                assert type(singleValue) == type(batchValue)
+                assert isinstance(singleValue, (float, Ethnicities))
+
+                filename = f"./{time()}.jpg"
+                msg = f"Batch estimation '{estimationType}' differs from single one. Saved as '{filename}'."
+                if isinstance(singleValue, Ethnicities):
+                    self.assertEth(expectedValue, singleValue)
+                    self.assertEth(expectedValue, batchValue)
+                else:  # age or gender
+                    assert expectedValue == int(singleValue) == int(batchValue), msg
+
+    def test_aggregation(self):
+        """
+        Test aggregation correctness.
+        """
+        estimations, aggregatedEstimation = self.estimateBatch(
+            [self._warp, self._warp2], estimateAge=True, estimateGender=True, estimateEthnicity=True, aggregate=True
+        )
+        for estimationType in ("Age", "Gender", "Ethnicity"):
+            estimationGetter: Callable[[Union[BasicAttributes, Eth]], Union[Ethnicities, float]] = attrgetter(
+                estimationType.lower()
+            )
+            with self.subTest(estimationType):
+                raw = list(map(estimationGetter, estimations))
+                aggregated = estimationGetter(aggregatedEstimation)
+                if isinstance(aggregated, Ethnicities):
+                    self.assertEth(Eth.fromSeveral(raw), aggregated)
+                else:  # age or gender
+                    assert int(mean(raw)) == int(aggregated)
+
+    def test_as_dict(self):
+        """
+        Test asDict method.
+        """
+        raw, aggregated = self.estimateBatch(
+            [self._warp, self._warp2], estimateAge=True, estimateGender=True, estimateEthnicity=True, aggregate=True
+        )
+        for estimation in [*raw, aggregated]:
+            assert isinstance(estimation.asDict(), dict)
```

Comment:
any negative test(s)?

Reply:
nope =) fsdk has only features, not bugs)